backend/src/main/java/saci/domain/service/LevelService.java

```diff
@@ -0,0 +1,35 @@
+package saci.domain.service;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import saci.domain.model.Level;
+import saci.domain.service.exceptions.AlreadyExistsException;
+import saci.domain.service.exceptions.CoefficientOverlapException;
+import saci.domain.service.validators.LevelValidator;
+import saci.infrastructure.LevelRepository;
+
+@Service
+@RequiredArgsConstructor
+public class LevelService {
+
+    private final LevelRepository levelRepository;
+
+    public Level createLevel(Level level) {
+        Optional<Level> optionalLevel = levelRepository.findByName(level.getName());
+        int overlappingLevels =
+                levelRepository.overlappingLevelsCounter(
+                        level.getRoleId(),
+                        Math.toIntExact(level.getMinCoefficient()),
+                        Math.toIntExact(level.getMaxCoefficient()));
+
+        if (!LevelValidator.LevelIsValid(level, overlappingLevels)) {
```

Comment:
```suggestion
        if (!LevelValidator.levelIsValid(level, overlappingLevels)) {
```

Just class names should be CapitalCased